rust/src/dataset/scanner.rs

```diff
@@ -197,6 +198,14 @@ impl Scanner {
         self
     }
 
+    /// Set whether to use the index if available
+    pub fn use_index(&mut self, use_index: bool) -> &mut Self {
```

Comment:
Can we design this api to be specific to ANN ?
in case there will be other secondary indices in the future.

Reply:
It's only used for ANN in the scanner code. It mimics refine_factor etc so can be refactored together when we add additional indices?